src/dll.py

```diff
@@ -76,6 +78,8 @@ def shift(self):
         if self.tail_node:
             self.tail_node.next_node = None
         self.length_list -= 1
+        if len(self) == 0:
```

Comment:
Again, it would be better to more assertively use a "Better to Beg Forgiveness than Ask Permission" approach. Checking for attributes that would only exist if the object is a Node.